Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/english/extractors/EnglishDateExtractorConfiguration.java

```diff
@@ -0,0 +1,206 @@
+    public static final Pattern DayRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.DayRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern MonthNumRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.MonthNumRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern YearRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.YearRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern WeekDayRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.WeekDayRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern SingleWeekDayRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.SingleWeekDayRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern OnRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.OnRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern RelaxedOnRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.RelaxedOnRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern ThisRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.ThisRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern LastDateRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.LastDateRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern NextDateRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.NextDateRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern DateUnitRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.DateUnitRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern SpecialDayRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.SpecialDayRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern WeekDayOfMonthRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.WeekDayOfMonthRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern RelativeWeekDayRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.RelativeWeekDayRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern SpecialDate = RegExpUtility.getSafeRegExp(EnglishDateTime.SpecialDate, Pattern.CASE_INSENSITIVE);
+    public static final Pattern SpecialDayWithNumRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.SpecialDayWithNumRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern ForTheRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.ForTheRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern WeekDayAndDayOfMonthRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.WeekDayAndDayOfMonthRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern RelativeMonthRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.RelativeMonthRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern PrefixArticleRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.PrefixArticleRegex, Pattern.CASE_INSENSITIVE);
+    public static final List<Pattern> DateRegexList = new ArrayList<Pattern>() {
+        {
+            add(RegExpUtility.getSafeRegExp(EnglishDateTime.DateExtractor1, Pattern.CASE_INSENSITIVE));
```

Comment:
We fixed all `Pattern.CASE_INSENSITIVE` that won't break tests. However, we found several regexs (like the commented one) that would break unit tests if the `Pattern.CASE_INSENSITIVE` is not specified.
We'll troubleshoot this and apply a fix in another PR since it seems this will impact in several classes.

Reply:
Perhaps the test code for Java DateTime was based on an older C# version. 
For Extractor and Parser tests, the expected vs actual text result should be compared in a case insensitive way. Or the specs for those need to have their results lowercased.
For models, everything should be OK.
We can merge with some patterns left as ```Pattern.CASE_INSENSITIVE``` and debug case by case and gradually remove them.

Reply:
Yes, that's the case. We based the entire porting to Java on an older version of C# (48cfdf94c4808f065bda19be15fb6dcd05af77a3). At some point, may be in parallel as we move forward, we will bring those new features and tests from the latest version of C# to the Java version.